.gitignore

```diff
@@ -1,16 +1,99 @@
-**/.DS_Store
+# <!-- Managed automatically by PreTeXt authoring tools -->
+# (delete the above line to manage this file manually)
+
+# Boilerplate list of files in a PreTeXt project for git to ignore
+# ensure this file is tracked
+!.gitignore
+
+# don't track unpublished builds or stage
 output
-schemas.xml
-*.tex
-*.swp
-generated-assets/webwork
+
+# don't track assets generated from source
+generated-assets
+
+# don't track the executables.ptx file
+executables.ptx
+
+# don't track node packages
+node_modules
+
+# don't track error logs
 .error_schema.log
-cli.log
+logs
+
+# don't track OS related files (windows/macos/linux)
+.DS_Store
+.DS_Store?
+._*
+.AppleDouble
+.LSOverride
+.Spotlight-V100
+.Trashes
+Icon
+.AppleDB
+.AppleDesktop
+Network Trash Folder
+Temporary Items
+.apdisk
+Thumbs.db
+Thumbs.db:encryptable
+ehthumbs.db
+ehthumbs_vista.db
+*.stackdump
+*.lnk
+*.cab
+*.msi
+*.msix
+*.msm
+*.msp
+[Dd]esktop.ini
+.directory
+.fuse_hidden*
+.Trash-*
+.nfs*
+
+# Don't include VSCode generated files
+.vscode
+*.code-workspace
+
+# Don't inlucde SublimeText files
+# Cache files for Sublime Text
+*.tmlanguage.cache
+*.tmPreferences.cache
+*.stTheme.cache
+
+# Workspace files are user-specific
+*.sublime-workspace
+
+# Project files should be checked into the repository, unless a significant
+# proportion of contributors will probably not be using Sublime Text
+*.sublime-project
+
+# SFTP configuration file
+sftp-config.json
+sftp-config-alt*.json
+
+# Package control specific files
+Package Control.last-run
+Package Control.ca-list
+Package Control.ca-bundle
+Package Control.system-ca-bundle
+Package Control.cache/
+Package Control.ca-certs/
+Package Control.merged-ca-bundle
+Package Control.user-ca-bundle
+oscrypto-ca-bundle.crt
+bh_unicode_properties.cache
+
+# Sublime-github package stores a github token in this file
+# https://packagecontrol.io/packages/sublime-github
+GitHub.sublime-settings
+
+
+# Don't include Dropbox settings and caches
+.dropbox
+.dropbox.attr
+.dropbox.cache
 
-Makefile.paths
-published
```

Comment:
This change is making my review of this PR a mess. At its core, I don't think I mind having `.gitignore` managed by the CLI. However, unless Runestone has stopped insisting that files be put into `published` rather than `output`, I think `published` also needs to be part of `.gitignore`. If CLI doesn't want to `.gitignore` the `published` directory, then we will continue manually maintaining our `.gitignore` file.

Reply:
I think that it's quite reasonable for us to add `published` to our default gitignore to support Runestone users: https://github.com/PreTeXtBook/pretext-cli/pull/770

But it's also fine to manually maintain your `.gitignore` if you prefer.

(I also think that @bnmnetp should consider using `output/runestone` instead of `published`. Or overriding `target.output_dir` during his build process to put the built files wherever he wants. But that's out of scope here!)

Reply:
> Its really not that hard to configure the project.ptx file for a runestone target to put the output in published.

I agree; here's the PR that shoud do this on Runestone's side so the next 50 authors don't have to take those extra steps to be compatible with Runestone. 🙃

https://github.com/RunestoneInteractive/rs/pull/494

(I don't have a RS dev environment handy so I haven't tested this.)